litestar/controller/generic.py

```diff
@@ -0,0 +1,226 @@
+# ruff: noqa: UP007, UP006
+from __future__ import annotations
+
+from inspect import isawaitable
+from operator import attrgetter
+from typing import TYPE_CHECKING, Any, Generic, List, Sequence, TypeVar, cast, get_args
+from uuid import UUID
+
+from typing_extensions import get_origin, get_type_hints
+
+from litestar.connection.request import Request
+from litestar.controller.base import Controller
+from litestar.enums import HttpMethod
+from litestar.exceptions import ImproperlyConfiguredException
+from litestar.handlers import BaseRouteHandler, HTTPRouteHandler
+from litestar.repository.filters import CollectionFilter
+from litestar.types import Empty
+from litestar.utils import is_class_and_subclass
+from litestar.utils.typing import get_safe_generic_origin
+
+if TYPE_CHECKING:
+    from litestar.dto import AbstractDTO
+    from litestar.repository import AbstractAsyncRepository, AbstractSyncRepository
+    from litestar.types import EmptyType
+
+ModelT = TypeVar("ModelT")
+IdAttrT = TypeVar("IdAttrT", str, int, UUID)
+
+GENERIC_METHOD_NAMES = (
+    "create_instance",
+    "update_instance",
+    "delete_instance",
+    "get_instance",
+    "create_many",
+    "update_many",
+    "delete_many",
+    "get_many",
+)
+
+
+class GenericController(Controller, Generic[ModelT, IdAttrT]):
+    repository_type: type[AbstractAsyncRepository[ModelT]] | type[AbstractSyncRepository[ModelT]]
+    """Repository for the controller's model."""
+    id_attribute: str = "id"
```

Comment:
The`id_attribute` is already defined on the repository. Can't that be used via `repository_type` 

I think it would be less ambiguous if it's defined once only.

Reply:
Ok, I'll look into it

Reply:
done